app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -411,12 +416,40 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         if (!browserOverlay.isVisible && session.isYoutubeTV &&
                 event.keyCode == KeyEvent.KEYCODE_BACK) {
             val escKeyEvent = KeyEvent(event.action, KeyEvent.KEYCODE_ESCAPE)
-            (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+
+            /**
+             * If YouTube guide-list element is focused:
```

Comment:
nit: the comments in here don't explain what problem we're trying to solve and, because our solution is complex, the solution at a high level. If I came across this code for the first time, I might be confused at why this BACK->ESC override is necessary (i.e. YouTube doesn't handle their back stack correctly) and why the override is only necessary _sometimes_.

For example, I might comment, "youtube/tv doesn't handle their back stack correctly: a user can navigate back by pressing ESC but going back in the browser back stack does nothing. To fix this, we convert BACK presses on youtube/tv to ESC. However, when we get to the top of YouTube's back stack, we back all the way out of YouTube."